MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,364 @@
+    try (Scope ignored = GlobalTracer.get().scopeManager().activate(span)) {
+      delegate.sendEmail(recipient, subject, body);
+    } finally {
+      span.finish();
+    }
+  }
+```
+
+After:
+```java
+@Override
+  public void sendEmail(To recipient, String subject, EmailBody body) {
+    var tracer = GlobalOpenTelemetry.get().getTracer(getClass().getName());
+    var span = tracer.spanBuilder("sendSmtpMail").startSpan();
+    try (var ignored = span.makeCurrent()) {
+      delegate.sendEmail(recipient, subject, body);
+    } finally {
+      span.end();
+    }
+  }
+```
+
+### Test Setup
+
+- Add the opentelemetry test dependency
+  ```gradle
+    testImplementation 'io.opentelemetry:opentelemetry-sdk-testing' 
+  ```
+
+- Replace `io.opentracing.mock.MockTracer` with `io.opentelemetry.sdk.testing.junit5.OpenTelemetryExtension`
+
+  Before:
+    ```java
+      private final MockTracer mockTracer = new MockTracer();
+    ```
+
+  After:
+    ```java
+      @RegisterExtension
+      @Order(0)
+      static final OpenTelemetryExtension OTEL = OpenTelemetryExtension.create();
+    ```
+
+- Clear metrics and spans before each test using OpenTelemetryExtension
+
+  Before:
+
+  ```java
+    @BeforeEach
+    void setupTestData() throws FolderException {
+    GlobalTracerTestUtil.setGlobalTracerUnconditionally(mockTracer);
+    mockTracer.reset();
+    }
+  ``` 
+  After:
+  ```java
+  @BeforeEach
+  void setupTestData() throws FolderException {
+    OTEL.clearMetrics(); // this is the OpenTelemetryExtension instance
+    OTEL.clearSpans();
+  }
+  ```
+
+- Capture spans using `OpenTelemetryExtension` and `io.opentelemetry.sdk.trace.data.SpanData`
+
+  Before:
+    ```java
+    assertThat(mockTracer.finishedSpans().stream()
+                            .filter(s -> s.operationName().equals("expectedTracing")));
+    ```
+  After:
+    ```java
+    assertThat(OTEL.getSpans().stream().filter(s -> s.getName().equals("expectedTracing")));
+    ```
+
+
+## Migrate from Morphia to Spring-Data-Mongo
+
+The spring-data-mongo package is available at the following coordinates:
+```groovy
+implementation "org.sdase.commons:sda-commons-server-spring-data-mongo"
+```
+
+### Replacing the MorphiaBundle:
+
+```diff
+-  private final MorphiaBundle<AppConfiguration> morphiaBundle =
+-    MorphiaBundle.builder()
+-      .withConfigurationProvider(AppConfiguration::getMongo)
+-      .withEntityScanPackageClass(ConsentConfigurationEntity.class)
+-      .withValidation()
+-      .build();
+
++  private final SpringDataMongoBundle<AppConfiguration> mongoBundle = 
++    SpringDataMongoBundle.builder()
++      .withConfigurationProvider(AppConfiguration::getMongo)
++      .build();
+```
+
+Morphia's `Datastore` can be replaced by Spring Data Mongo's `MongoOperations`:
+```java
+public MongoOperations getMongoOperations() {
+  return this.mongoBundle.getMongoOperations();
+}
+```
+
+#### Morphia compatibility
```

Comment:
let's use level 3 indentation for all subheaders.

Reply:
updated